library/psa_crypto_pake.c

```diff
@@ -0,0 +1,720 @@
+            default:
+                return( PSA_ERROR_BAD_STATE );
+        }
+
+        operation->sequence = PSA_PAKE_X1_STEP_KEY_SHARE;
+    }
+
+    /* Check if step matches current sequence */
+    switch( operation->sequence )
+    {
+        case PSA_PAKE_X1_STEP_KEY_SHARE:
+        case PSA_PAKE_X2_STEP_KEY_SHARE:
+            if( step != PSA_PAKE_STEP_KEY_SHARE )
+                return( PSA_ERROR_BAD_STATE );
+            break;
+
+        case PSA_PAKE_X1_STEP_ZK_PUBLIC:
+        case PSA_PAKE_X2_STEP_ZK_PUBLIC:
+            if( step != PSA_PAKE_STEP_ZK_PUBLIC )
+                return( PSA_ERROR_BAD_STATE );
+            break;
+
+        case PSA_PAKE_X1_STEP_ZK_PROOF:
+        case PSA_PAKE_X2_STEP_ZK_PROOF:
+            if( step != PSA_PAKE_STEP_ZK_PROOF )
+                return( PSA_ERROR_BAD_STATE );
+            break;
+
+        default:
+            return( PSA_ERROR_BAD_STATE );
+    }
+
+    /* Initialize & write round on KEY_SHARE sequences */
+    if( operation->state == PSA_PAKE_OUTPUT_X1_X2 &&
+        operation->sequence == PSA_PAKE_X1_STEP_KEY_SHARE )
+    {
+        ret = mbedtls_ecjpake_write_round_one( &operation->ctx.ecjpake,
+                                               operation->buffer,
+                                               PSA_PAKE_BUFFER_SIZE,
+                                               &operation->buffer_length,
+                                               mbedtls_psa_get_random,
+                                               MBEDTLS_PSA_RANDOM_STATE );
+        if( ret != 0 )
+        {
+            psa_pake_abort( operation );
+            return( mbedtls_to_psa_error( ret ) );
+        }
+
+        operation->buffer_offset = 0;
+    }
+    else if( operation->state == PSA_PAKE_OUTPUT_X2S &&
+             operation->sequence == PSA_PAKE_X1_STEP_KEY_SHARE )
+    {
+        ret = mbedtls_ecjpake_write_round_two( &operation->ctx.ecjpake,
+                                               operation->buffer,
+                                               PSA_PAKE_BUFFER_SIZE,
+                                               &operation->buffer_length,
+                                               mbedtls_psa_get_random,
+                                               MBEDTLS_PSA_RANDOM_STATE );
+        if( ret != 0 )
+        {
+            psa_pake_abort( operation );
+            return( mbedtls_to_psa_error( ret ) );
+        }
+
+        operation->buffer_offset = 0;
+    }
+
+    /* Load output sequence length */
+    if( operation->state == PSA_PAKE_OUTPUT_X2S &&
+        operation->sequence == PSA_PAKE_X1_STEP_KEY_SHARE )
+    {
+        if( operation->role == PSA_PAKE_ROLE_SERVER )
+            /* Length is stored after 3bytes curve */
+            length = 3 + operation->buffer[3] + 1;
+        else
+            /* Length is stored at the first byte */
+            length = operation->buffer[0] + 1;
+    }
+    else
+    {
+        /* Length is stored at the first byte */
```

Comment:
The comment does not match to the code.

Reply:
It does, because it's the "first byte" of the current chunk, I'll update the comment